### DISCUSSION TITLE
How to make GTK apps use the  status notifier panel widget instead of the now obsolete system tray widget?

### DISCUSSION THREAD

**Discussion by renegat73** [lxqt/lxqt#1987] [Q&A]
![lxqt_icon_problem](https://user-images.githubusercontent.com/66755519/116812491-472f9980-ab4f-11eb-91a1-882e003c019b.png)

![20210502-135345](https://user-images.githubusercontent.com/66755519/116812510-5ca4c380-ab4f-11eb-95d8-b33699c2725b.png)

It is still the old [problem:](https://github.com/lxqt/lxqt/issues/1864) Some GTK+ applications icons appear blurry and without function in the lxqt panels 'system tray' widget.

This issue was said to be solved with LXQt 0.17 by using 'status notifier' widget instead of 'system tray'.
The older 'system tray' widget is now marked as obsolete with the hint to use 'status notifier' plugin instead.

But how can I make the affected applications (claws-mail and blueman-manager) use the 'status notifier' plugin? The only app that appears in the configuration of status notifier is redshift-qt.

Without the now obsolete 'system tray' plugin there will be no icons for blueman and claws-mail at all.

I was not able to find any documentatin about the new functionality of 'status notifier'.

**Comment by tsujan** [comment]
If you have libappindicator-gtk3 and libappindicator-gtk2, blueman-manager will appear in the status notifier.

As for claws-mail, you may want to remind its devs that it's 2021.

**Reply by renegat73** [reply]
Thank you for pointing me to libappindicator-gtk3 and libappindicator-gtk2.
Perhaps these should be installed along with lxqt-panel as (optional) dependencies by the distribution (archlinux for me).

Bluetooth-icon now works as expected, claws-mail still does not make use of statusnotifier.

I'd already reported this issue to claws-mail bug tracker half a year ago but no one seems to care about it.
https://www.thewildbeast.co.uk/claws-mail/bugzilla/show_bug.cgi?id=4409

Perhaps it would be the best solution to simply change the local mail-client program - but afaik there is still no native Qt mail application beside trojita which unfortunately only supports IMAP but not POP3.

**Reply by tsujan** [reply]
I only use Thunderbird. Yes, it's heavy but LXQt is light in order to give the resources to other apps.

[Birdtray](https://github.com/gyunaev/birdtray) is the tray icon for Thunderbird. It's written in Qt.

**Comment by tsujan** [comment]
> Perhaps these should be installed along with lxqt-panel as (optional) dependencies by the distribution (archlinux for me).

Yes, that's a good idea.

However, please also note that the future of tray icons of GTK apps is suspicious. Considering the way Gnome devs change GTK, I won't be surprised if no GTK app could have a tray icon at some point.

**Comment by elviosak** [comment]
there's this https://aur.archlinux.org/packages/xembed-sni-proxy-git/ it "converts" tray icons into StatusNotifierItems.
it's part of KDE, so it might be available if u install it (haven't checked) but if not u can install from AUR, there's a small problem where it will update (when checking with AUR helper) when there's any change on the repo (there's a lot more stuff there).
